manim/utils/hashing.py

```diff
@@ -285,8 +286,11 @@ def _iter_check_dict(dct):
             return _iter_check_list(iterable)
         elif isinstance(iterable, dict):
             return _iter_check_dict(iterable)
+        else:
+            # mypy requires this line, even though it should not be reached.
+            return iterable
```

Comment:
MyPy requires this line, because, in the hypothetical case that `_cleaned_iterable()` receives a different object that's neither an iterable or a dictionary, the function would implicitly return `None`.

Now, as you mention, this shouldn't be reached, because the object should always be an iterable or a dictionary. In this case, I suggest raising an exception instead of silently returning the same iterable. In this way, if `_cleaned_iterable()` is being passed something different, we can catch the bug instead of silently omitting it:
```suggestion
            raise TypeError("'iterable' is neither an iterable nor a dictionary.")
```